ckanext/example_idatadictionaryform/plugin.py

```diff
@@ -0,0 +1,120 @@
+# encoding: utf-8
+
+from __future__ import annotations
+
+from typing import Any, cast
+from ckan.types import Schema, ValidatorFactory
+from ckan.common import CKANConfig
+from ckan.lib.navl.dictization_functions import missing
+from ckan.types import (
+    Context, FlattenDataDict, FlattenErrorDict, FlattenKey,
+)
+
+import json
+
+from ckan.plugins.toolkit import (
+    Invalid, get_validator, add_template_directory, _
+)
+from ckan import plugins
+from ckanext.datastore.interfaces import IDataDictionaryForm
+
+
+class ExampleIDataDictionaryFormPlugin(plugins.SingletonPlugin):
+    plugins.implements(IDataDictionaryForm)
+    plugins.implements(plugins.IConfigurer)
+
+    # IConfigurer
+
+    def update_config(self, config: CKANConfig):
+        add_template_directory(config, 'templates')
+
+    # IDataDictionaryForm
+
+    def update_datastore_create_schema(self, schema: Schema):
+        ignore_empty = get_validator('ignore_empty')
+        int_validator = get_validator('int_validator')
+        unicode_only = get_validator('unicode_only')
+        to_datastore_plugin_data = cast(
+            ValidatorFactory, get_validator('to_datastore_plugin_data'))
+        to_eg_iddf = to_datastore_plugin_data('example_idatadictionaryform')
+
+        f = cast(Schema, schema['fields'])
+        f['an_int'] = [ignore_empty, int_validator, to_eg_iddf]
+        f['json_obj'] = [ignore_empty, json_obj, to_eg_iddf]
+        f['only_up'] = [
+            only_increasing, ignore_empty, int_validator, to_eg_iddf]
+        f['sticky'] = [
+            default_current, ignore_empty, unicode_only, to_eg_iddf]
+
+        # use different plugin_key so that value isn't removed
+        # when above fields are updated & value not exposed in
+        # datastore_info
+        f['secret'] = [
+            ignore_empty,
+            to_datastore_plugin_data('example_idatadictionaryform_secret')
+        ]
+        return schema
+
+    def update_datastore_info_field(
+            self, field: dict[str, Any], plugin_data: dict[str, Any]):
+        # expose all our non-secret plugin data in the field
```

Comment:
Yes a plugin chooses how or whether to expose its data in the fields dict. Once case where it makes sense to hide the data stored is for statistics captured by automatic analysis that may be used for indexing but may be too large or rarely needed by users.

I did consider adding a parameter to `datastore_info` that could be passed through to `update_datastore_info_field` so that a user could request specific data to be included in the output. Let me know if you think that should be included here.